system/src/system_network_wifi.h

```diff
@@ -223,6 +236,11 @@ class WiFiNetworkInterface : public ManagedIPNetworkInterface<WLanConfig, WiFiNe
         return wlan_get_hostname(buf, buf_len, NULL);
     }
 
+private:
+    // It's tricky to detect the modem power state.
+    // We assume that every time the device being powered up, the module is powered on.
+    // The the power state then be tracked by the on_now() and off_now() APIs.
+    bool powered = true;
```

Comment:
I think we previously figured out that on boot BCM43362 is powered off, so we should be able to safely set this to `false` by default?

Reply:
I think it's on by default: https://github.com/particle-iot/device-os/pull/2205#discussion_r529387030